src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -147,6 +148,11 @@ private function populateAvailableType($id, Definition $definition)
             $this->types[$type] = $id;
         }
 
+        // Cannot use reflection if the class isn't set
```

Comment:
It depends of the class of the definition. It cannot be moved outside of the loop.